consensus/src/consensus/consensus_proxy.rs

```diff
@@ -94,26 +95,58 @@ impl<N: Network> ConsensusProxy<N> {
 
             match response {
                 Ok(response) => {
-                    // Group transaction hashes by epoch to reduce number of requested proofs
-                    let mut hashes_by_epoch = HashMap::new();
+                    let blockchain = self.blockchain.read();
+
+                    // Group transaction hashes by the block number that proofs those transactions to reduce the number of requests
+                    // There are three categories of block numbers:
+                    //  - Finalized epochs: we use the election block number that finalized the respective epoch
+                    //  - Finalized batch in the current epoch: We use the latest checkpoint block number
+                    //  - Current batch: We use the current head to prove those transactions
+
+                    // This is the structure where we group transactions by their proving block number
+                    let mut hashes_by_block = HashMap::new();
+
+                    let election_head_number = blockchain.election_head().block_number();
+
+                    let checkpoint_head_number = blockchain.macro_head().block_number();
+
+                    let current_head_number = blockchain.head().block_number();
```

Comment:
I'd remove the extra lines
```suggestion
                    // This is the structure where we group transactions by their proving block number
                    let mut hashes_by_block = HashMap::new();
                    let election_head_number = blockchain.election_head().block_number();
                    let checkpoint_head_number = blockchain.macro_head().block_number();
                    let current_head_number = blockchain.head().block_number();
```